shared/teams/team/container.tsx

```diff
@@ -5,34 +5,30 @@ import Team, {Props} from '.'
 import CustomTitle from './custom-title/container'
 import {HeaderRightActions, HeaderTitle, SubHeader} from './nav-header/container'
 import * as Kb from '../../common-adapters'
-import {connect, compose, getRouteProps, isMobile} from '../../util/container'
+import * as Container from '../../util/container'
 import * as Constants from '../../constants/teams'
 import {mapStateHelper as invitesMapStateHelper, getRows as getInviteRows} from './invites-tab/helper'
 import {mapStateHelper as memberMapStateHelper, getRows as getMemberRows} from './members-tab/helper'
 import {mapStateHelper as subteamsMapStateHelper, getRows as getSubteamsRows} from './subteams-tab/helper'
 import {RouteProps} from '../../route-tree/render-route'
 
-type OwnProps = RouteProps<
-  {
-    teamname: string
-  },
-  {}
-> & {
+type OwnProps = Container.RouteProps<{teamname: string}> & {
   selectedTab: string
   setSelectedTab: (arg0: string) => void
 }
 
 // keep track during session
 const lastSelectedTabs = {}
 
-const mapStateToProps = (state, ownProps: OwnProps) => {
-  const teamname = getRouteProps(ownProps, 'teamname')
+const mapStateToProps = (state: Container.TypedState, ownProps: OwnProps) => {
+  const teamname = Container.getRouteProps<string>(ownProps, 'teamname')
   if (!teamname) {
     throw new Error('There was a problem loading the team page, please report this error.')
   }
 
   const selectedTab = ownProps.selectedTab || 'members'
 
+  // TODO this shouldn't be implemented as smashing these unrelated states together makes the typing really messy and this hard to understand
```

Comment:
the typing here is nasty as this connector does the work of 2 unrelated things. it really should be split up